### DISCUSSION TITLE
Passing data from parent to child widget through a provider

### DISCUSSION THREAD

**Discussion by fujidaiti** [rrousselGit/riverpod#1698] [Q&A] [answered]
I’ve read the example of todo app, and found the following code:
https://github.com/rrousselGit/riverpod/blob/5a454426adc01c660b3c9b566d447ec0dfe25c70/examples/todos/lib/main.dart#L119-L124
Task data is passed from the parent ListView to its child through the provider `_currentTodo` using ProviderScope,  and `_currentTodo` is defined as:
https://github.com/rrousselGit/riverpod/blob/5a454426adc01c660b3c9b566d447ec0dfe25c70/examples/todos/lib/main.dart#L225-L232
I understand what the comment says, 

> /// By retrieving the [Todo] through a provider instead of through its
> /// constructor, this allows [TodoItem] to be instantiated using the `const` keyword.

but Is this a common practice in riverpod? It seems a little bit tricky to me.

**Comment by rrousselGit** [comment, marked as answer]
It's a performance optimization. It's optional

**Reply by fujidaiti** [reply]
thank you😀
